### PR TITLE
Support Flyway ignoreMissingMigrations config

### LINKED ISSUE #11421: Add support for Flyway "missing migrations"
**Description**
Flyway can be configured to ignore missing migrations (see `FluentConfiguration#ignoreMissingMigrations(boolean)` for details). This configuration is not currently exposed by the Quarkus Flyway extension.

**Implementation ideas**
Add a new boolean-typed `quarkus.flyway.ignore-missing-migrations` configuration property (defaulting to `false` as in Flyway).

### PR DESCRIPTION
**Pull request by knutwannheden**
Fixes #11421

### CODE REVIEW

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ void testValidateOnMigrate() {
 
     @Test
     @DisplayName("outOfOrder is correctly set")
-    void testOutOfOrder() {
+    void testIgnoreMissing() {
```
Comment (gsmet) [inline extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java:172]: There is something wrong here.

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java [hunk 2]
```diff
@@ -179,6 +179,18 @@ void testOutOfOrder() {
         assertTrue(createdFlywayConfig().isOutOfOrder());
     }
 
+    @Test
+    @DisplayName("ignoreMissingMigrations is correctly set")
+    void testOutOfOrder() {
```
Comment (gsmet) [inline extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java:184]: And then here. The test names are inconsistent.

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 3]
```diff
@@ -108,6 +108,15 @@ public static final FlywayDataSourceRuntimeConfig defaultConfig() {
     @ConfigItem
     public boolean outOfOrder;
 
+    /**
+     * Ignore missing migrations when reading the history table. When set to true any migration present in the history table but
+     * absent in the configured locations will be ignored (and logged as a warning), when false (the default) the validation
+     * step
+     * will fail.
```
Comment (gsmet) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:115]: Very minor but could you put `step` and `will fail` on the same line?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks! I added some minor comments, could you adjust and squash everything in one commit?

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java (hunk 3 above)
Comment (knutwannheden) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:115]: Ah, I think it must have been the formatter (during the build) which changed this. I need to check my IntelliJ IDE settings.

**Comment by knutwannheden** [comment]
Thanks for the review. I pushed another commit to fix the issues. If you don't want to squash the commits as part of the merging of the PR, I can of course also squash them locally and force-push it on this branch, although I suspect that the review comments might get lost then.

**Comment by knutwannheden** [comment]
It does seem like you typically squash locally and force-push PRs in the Quarkus project, so I went ahead and did that now.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@knutwannheden yeah, we do merge commits and we can't squash with the GH UI when doing so. Thanks!

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**